modelservice/src/main/java/com/expedia/adaptivealerting/modelservice/dao/es/ElasticSearchDetectorMappingService.java

```diff
@@ -0,0 +1,377 @@
+                new PercolateQueryBuilder(DetectorMappingEntity.QUERY_KEYWORD, elasticSearchConfig.getDocType(),
+                    refList, XContentType.JSON);
+            SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder();
+            ConstantScoreQueryBuilder constantScoreQueryBuilder = new ConstantScoreQueryBuilder(percolateQuery);
+            searchSourceBuilder.query(constantScoreQueryBuilder);
+            searchSourceBuilder.timeout(new TimeValue(elasticSearchConfig.getConnectionTimeout()));
+            //FIXME setting default result set size to 500.
+            searchSourceBuilder.size(500);
+            return getDetectorMappings(searchSourceBuilder, tagsList);
+        } catch (IOException e) {
+            this.exceptionCount.inc();
+            log.error("Error ES lookup", e);
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public String createDetectorMapping(CreateDetectorMappingRequest createRequest) {
+        Set<String> fields = getFields(createRequest);
+        Set<String> newFieldMappings = getFieldsWithoutExistingMapping(fields);
+        updateIndexMappings(newFieldMappings);
+        final DetectorMappingEntity detectorMappingEntity = new DetectorMappingEntity(createRequest.getUser(),
+            createRequest.getDetector(), QueryUtil.buildQuery(createRequest.getExpression()), true,
+            System.currentTimeMillis(), System.currentTimeMillis());
+        final IndexRequest indexRequest = new IndexRequest(elasticSearchConfig.getIndexName(),
+            elasticSearchConfig.getDocType());
+        indexRequest.source(DetectorMappingEntityToJson(detectorMappingEntity), XContentType.JSON);
+        try {
+            IndexResponse indexResponse = elasticSearchClient.index(indexRequest, RequestOptions.DEFAULT);
+            return indexResponse.getId();
+        } catch (IOException e) {
+            log.error(String.format("Indexing mapping %s failed", DetectorMappingEntityToJson(detectorMappingEntity), e));
+            throw new RuntimeException(e);
+        }
+    }
+    
+    private void updateIndexMappings(Set<String> newFieldMappings) {
+        PutMappingRequest request = new PutMappingRequest(elasticSearchConfig.getIndexName());
+        Map<String, Object> jsonMap = new HashMap<>();
+        Map<String, Object> type = new HashMap<>();
+        type.put("type", "keyword");
+        Map<String, Object> properties = new HashMap<>();
+        newFieldMappings.forEach(field -> {
+            properties.put(field, type);
+        });
+        jsonMap.put("properties", properties);
+        request.source(jsonMap);
+        request.type(elasticSearchConfig.getDocType());
+        try {
+            elasticSearchClient.indices().putMapping(request, RequestOptions.DEFAULT);
+        } catch (IOException e) {
+            log.error("Error updating mappings", e);
+            throw new RuntimeException(e);
+        }
+    }
```

Comment:
Check out the following, in which I've reordered some of the statements and used line breaks to clarify the logic:

```
    private void updateIndexMappings(Set<String> newFieldMappings) {
        Map<String, Object> type = new HashMap<>();
        type.put("type", "keyword");

        Map<String, Object> properties = new HashMap<>();
        newFieldMappings.forEach(field -> {
            properties.put(field, type);
        });

        Map<String, Object> jsonMap = new HashMap<>();
        jsonMap.put("properties", properties);
        
        PutMappingRequest request = new PutMappingRequest(elasticSearchConfig.getIndexName());
        request.source(jsonMap);
        request.type(elasticSearchConfig.getDocType());
        
        try {
            elasticSearchClient.indices().putMapping(request, RequestOptions.DEFAULT);
        } catch (IOException e) {
            log.error("Error updating mappings", e);
            throw new RuntimeException(e);
        }
    }
```

In general it's easier to understand the code when instance creation and property setting are together in the code. Consider adopting the above.

Reply:
Fixed.